File: test-junit5/src/test/java/io/micronaut/test/junit5/PropertyValueTest.java

package io.micronaut.test.junit5;

import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.MethodOrderer.OrderAnnotation;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestMethodOrder;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
@Property(name = "foo.bar", value = "stuff")
@TestMethodOrder(OrderAnnotation.class)
class PropertyValueTest {

    @Property(name = "foo.bar")
    String val;

    @Property(name = "prop.from.yml")
    String fromYml;

    @Test
    @Order(1)
    void testInitialValue() {
        assertEquals("stuff", val);
    }

    @Property(name = "foo.bar", value = "changed")
    @Test
    @Order(2)
    void testValueChanged() {
        assertEquals("changed", val);
    }

    @Test
    @Order(3)
    void testValueRestored() {
        assertEquals("stuff", val);
    }

    @Test
    @Property(name = "prop.from.yml", value = "local")
    @Order(4)
    void testValueOverridenFromConfig() {
        assertEquals("local", fromYml);
    }

    @Test
    @Order(5)
    void testValueFromConfig() {
        assertEquals("yml", fromYml);
    }
}

File: test-suite-python/src/test/python/micronaut/test/python/PropertyValueTest.py
from typing import Annotated

from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest
@Property(name="foo.bar", value="stuff")
class PropertyValueTest:

    val: Annotated[str, Property(name="foo.bar")]

    @Test
    def test_initial_value(self):
        assert "stuff" == self.val

    @Property(name="foo.bar", value="changed")
    @Test
    def test_value_changed(self):
        assert "changed" == self.val
